assets/src/design-system/components/radio/radio.js

```diff
@@ -0,0 +1,193 @@
+    pointer-events: none;
+    display: none;
+
+    :after {
+      content: '';
+      position: absolute;
+      height: ${RADIO_DIAMETER}px;
+      width: ${RADIO_DIAMETER}px;
+      top: 50%;
+      left: 50%;
+      transform: translate(-50%, -50%);
+      background-color: ${theme.colors.fg.primary};
+      border-radius: ${theme.borders.radius.round};
+    }
+  `
+);
+
+const HiddenInput = styled.input.attrs({ type: 'radio' })`
+  position: absolute;
+  height: ${BUTTON_CONTAINER_HEIGHT}px;
+  width: ${BUTTON_CONTAINER_HEIGHT}px;
+  margin: 0;
+  opacity: 0;
+  cursor: pointer;
+
+  :hover ~ ${RadioBorder} {
+    border-color: ${({ theme }) => theme.colors.border.defaultHover};
+  }
+
+  :focus {
+    outline: none;
+  }
+
+  :focus ~ ${RadioBorder}, &.${FOCUS_VISIBLE_SELECTOR} ~ ${RadioBorder} {
+    ${({ theme }) => themeHelpers.focusCSS(theme.colors.border.focus)};
+    border-color: ${({ theme }) => theme.colors.border.defaultHover};
+  }
+
+  :active ~ ${RadioBorder} {
+    box-shadow: 0 0 0 8px ${({ theme }) => theme.colors.shadow.active};
+  }
+
+  :checked ~ ${InnerButton} {
+    display: block;
+  }
+
+  :disabled {
+    ~ ${RadioBorder} {
+      border-color: ${({ theme }) => theme.colors.border.disable};
+    }
+
+    ~ ${InnerButton}::after {
+      background-color: ${({ theme }) => theme.colors.fg.disable};
+    }
+  }
+`;
+
+const Hint = styled(Text).attrs({
+  forwardedAs: 'span',
+  size: THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL,
+})`
+  color: ${({ theme }) => theme.colors.fg.secondary};
+`;
+
+/**
+ * A controlled radio button. Can be used with other radio buttons.
+ *
+ * To configure a group of radio buttons, the following must be done:
+ * 1. Give all the radio buttons of a group the same `name`
+ * 2. Pass a unique `value` to each radio button. This tracks the value of the specific radio button.
+ * 3. Pass `checked={true}` to the radio button that should be selected
+ *
+ * Keyboard navigation:
+ * - [Tab] can be used to navigate between radio groups
+ * - [ArrowUp] and [ArrowDown] can be used to navigate between radio buttons
+ *
+ * @param {Object} props The props
+ * @param {string} props.className className that is applied to the outer container
+ * @param {string} props.checked whether the radio button is selected
+ * @param {string} props.hint hint to display
+ * @param {string} props.id unique id used to associate the label to the input
+ * @param {string} props.label label to display
+ * @param {Function} props.onChange change event handler
+ * @param {string} props.value the value of the radio button
+ * @return {Object} The radio button
+ */
+export function Radio({ className, hint, id, label, ...props }) {
+  const inputId = useMemo(() => id || uuidv4(), [id]);
+
+  return (
+    <Container className={className}>
+      <ButtonContainer>
+        <HiddenInput {...props} />
+        <RadioBorder />
+        <InnerButton />
+      </ButtonContainer>
+
+      {(label || hint) && (
```

Comment:
Maybe the label should be required for a radio -- can't think of a use case right now where the radio alone would be needed. Thoughts?

Reply:
Sure I can add that! If it needs to be optional in the future we can always come back to it